### PR TITLE
style(incarnation-scafi): add scalafmt as scala code linter

### PR DESCRIPTION
**Pull request by cric96**

### CODE REVIEW

**Comment by DanySK** [comment]
It looks like you are having some struggles with rebase :smile: 

You can merge, I'll squash that in anyway.

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    RedundantParens,
+    SortModifiers
+  ]
+  redundantBraces.maxLines = 1
+  redundantBraces.stringInterpolation = true
+}
+
+includeNoParensInSelectChains = false
+
+optIn.blankLineBeforeDocstring = true
```
Comment (DanySK) [inline .scalafmt.conf:41]:
```suggestion
optIn.blankLineBeforeDocstring = true

```

(newline at end of file)

File: alchemist-incarnation-scafi/src/test/scala/it/unibo/alchemist/scafi/test/TestInSimulator.scala [hunk 2]
```diff
@@ -8,14 +8,12 @@
...
- * This file is part of Alchemist, and is distributed under the terms of the
- * GNU General Public License, with a linking exception, as described in the file
- * LICENSE in the Alchemist distribution's top directory.
- ******************************************************************************/
+/** ***************************************************************************** Copyright (C) 2010-2018, Danilo
+  * Pianini and contributors listed in the main project's alchemist/build.gradle file.
+  *
+  * This file is part of Alchemist, and is distributed under the terms of the GNU General Public License, with a linking
+  * exception, as described in the file LICENSE in the Alchemist distribution's top directory.
+  */
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/test/scala/it/unibo/alchemist/scafi/test/TestInSimulator.scala:16]: Looks out of date...

File: settings.gradle.kts [hunk 3]
```diff
@@ -49,6 +49,7 @@ gitHooks {
     commitMsg { conventionalCommits() }
     preCommit {
         tasks("ktlintCheck")
+        tasks("checkScalafmt")
```
Comment (DanySK) [inline settings.gradle.kts:52]: How long does it take?

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -109,3 +109,4 @@ publishOnCentral = "org.danilopianini.publish-on-central:0.7.19"
 scalatest = "com.github.maiflai.scalatest:0.32"
 shadowJar = "com.github.johnrengelman.shadow:7.1.2"
 taskTree = "com.dorongold.task-tree:2.1.0"
+scalafmt = "cz.alenkacz.gradle.scalafmt:1.16.2"
```
Comment (DanySK) [inline gradle/libs.versions.toml:112]: must be in alphabetical order, and there should be a newline at end of file

**Review by DanySK — CHANGES_REQUESTED** [review]
Few minors.

File: settings.gradle.kts (hunk 3 above)
Comment (cric96) [reply settings.gradle.kts:52]: 3 seconds in my pc: https://scans.gradle.com/s/ze7zskbahomi6/timeline#ynxm2bmfq4cim

Comment (DanySK) [reply settings.gradle.kts:52]: okay then. I see it does not depend on `compileScala` which is too slow

**Pull request merged**